### PR TITLE
Add reader for NWCSAF/PPS which can also be used by NWCSAF/MSG

### PR DESCRIPTION
**Pull request by adybbroe**
Added a yaml-reader plugin for NWCSAF/PPS cloud mask and type.
Added a common reader or msg & pps products. 

Remaining issues: The NWCSAF-MSG yaml file is still using the msg-reader, but should be tested using the new common reader, and then removed when tested successful. Also, the remaining PPS products should be added.

Signed-off-by: Adam.Dybbroe <a000680@c20671.ad.smhi.se>

### CODE REVIEW

File: satpy/etc/readers/nc_nwcsaf_pps.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+reader:
+  description: NetCDF4 reader for the NWCSAF/PPS 2014 format
+  name: nc_nwcsaf
+  sensors: ['avhrr-3', 'viirs']
```
Comment (mraspaud) [inline satpy/etc/readers/nc_nwcsaf_pps.yaml]: How about modis ?

File: satpy/readers/nc_nwcsaf.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        proj = Dataset(np.squeeze(values),
+                       copy=False,
+                       **info)
+        return proj
+
+    def get_area_def(self, dsid):
+        """Get the area definition of the datasets in the file. 
+        Only applicable for MSG products!"""
+        if self.pps:
+            # PPS:
```
Comment (mraspaud) [inline satpy/readers/nc_nwcsaf.py]: Is that enough to make sure it's not gridded data ? don't we have reprojected nwcsaf pps products ?

File: satpy/etc/readers/nc_nwcsaf_pps.yaml (hunk 1 above)
Comment (adybbroe) [inline satpy/etc/readers/nc_nwcsaf_pps.yaml]: Yes, modis should be added.

File: satpy/readers/nc_nwcsaf.py (hunk 2 above)
Comment (adybbroe) [inline satpy/readers/nc_nwcsaf.py]: Ahh, yes, you are right in a way. Though the mapped netCDF pps products are not the offical/standard PPS output. I would suggest we only support the swath products for the moment.

Comment (mraspaud) [inline satpy/readers/nc_nwcsaf.py]: Ok, can we add this to the documentation then as limitation of the nwcsaf_pps reader?

Comment (adybbroe) [inline satpy/readers/nc_nwcsaf.py]: Done. Also added more products.

**Pull request merged**